### PR TITLE
Add Firewall Documentation

### PR DESCRIPTION
**Pull request by lkiesow**
This patch adds a simple overview of network communication in an
Opencast cluster and suggests two simple configurations.

---

The graphviz renders as:
network

### Your pull request should…

* [x] have a concise title
* [x] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [x] be against the correct branch (features can only go into develop)
* [x] include migration scripts and documentation, if appropriate
* [x] pass automated tests
* [x] have a clean commit history
* [x] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

### CODE REVIEW

**Comment by Arnei** [comment]
I successfully build this (according to the instructions in `docs/guides/readme.md`) , but the graphic does not show. Is that to be expected?
Bildschirmfoto vom 2020-12-18 10-52-25

**Comment by lkiesow** [comment]
Ups, there seems to be one dependency missing from the docs. Try:

```sh
pip install mkdocs mkdocs-windmill
pip install markdown_inline_graphviz_extension
pip install markdown-inline-graphviz-extension-png
```

If I remember correctly, you had to run them as separate commands due to setup whankyness.
Will fix the docs…

---

EDIT: [Docs are fixed](https://github.com/opencast/opencast/tree/93ccb8d1fd915d4e6871d69806a9eb8b5415f158/docs/guides#quick-build)

**Review by Arnei — APPROVED** [review]

File: docs/guides/admin/docs/configuration/firewall.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+This describes a lot of the communication happening between servers in the most common use cases.
+In certain set-ups and with certain types of integrations, these may deviate slightly.
+
+General rules are:
+
+- Users communicate with Opencast via HTTP(S)
+- Capture agents communicate with Opencast via HTTP(S). All communication is established by the agent.
+- Opencast nodes communicate among each other via HTTP(S)
+- Often Elasticsaecrh and ActiveMQ are run on the admin node since most communication then happens on the server
```
Comment (gregorydlogan) [inline docs/guides/admin/docs/configuration/firewall.md]: Elastics**ea**rch

File: docs/guides/admin/docs/configuration/firewall.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    - The admin node communicates with Elasticsaecrh exclusively
+    - Some messages are sent to ActiveMQ from other nodes
+- All servers should get storage access
```
Comment (gregorydlogan) [inline docs/guides/admin/docs/configuration/firewall.md]: `network storage access` perhaps?  Something to make it clear that something like NFS is required.

File: docs/guides/admin/docs/configuration/firewall.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+2. Allow all TCP communication within the cluster
+
+
+### Complex
+
+If you want a more complex, stricter set of rules:
+
+1. Allow external HTTP and HTTPS communication to admin, presentation and possibly ingest
+2. Allow all Opencast nodes to access ActiveMQ and the database
+3. Allow the admin node to access Elasticsearch
```
Comment (gregorydlogan) [inline docs/guides/admin/docs/configuration/firewall.md:83]: 4. Allow all nodes access to the storage infrastructure ?
I get that you're deliberately excluding other services (cf SSH), but I can see how people might be confused here if we don't mention it.

File: docs/guides/admin/docs/configuration/firewall.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+Opencast relies on a lot of network communication and although not strictly necessary,
+it usually makes sense to configure a firewall for additional protection
+
+This describes a lot of the communication happening between servers in the most common use cases.
+In certain set-ups and with certain types of integrations, these may deviate slightly.
+
+General rules are:
+
+- Users communicate with Opencast via HTTP(S)
+- Capture agents communicate with Opencast via HTTP(S). All communication is established by the agent.
```
Comment (gregorydlogan) [inline docs/guides/admin/docs/configuration/firewall.md]: This is the only item with periods.  The lines in the diagram are already directional, maybe omit the last sentence.  Or add periods to all of them :)

**Review by gregorydlogan — CHANGES_REQUESTED** [review]

**Comment by lkiesow** [comment]
Fixed another typo. Ready for re-review

**Pull request merged**